services/CustomFees.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import "BasicTypes.proto";
+import "google/protobuf/wrappers.proto";
+
+/* A fraction of the units to be collected from the payer of a CryptoTransfer 
+that transfers units of a token. The amount charged will never be
+less than the stated minimumUnitsToCollect, and never greater than 
+the stated maximumUnitsToCollect (if given). */
+message FractionalFee {
+  Fraction fractionOfUnitsToCollect = 1; // percentage of the Tokens involved in the transfer to be charged as fee
```

Comment:
This is inefficient to transfer and store, including for the mirror node where we have to store it indefinitely. Why can't we just use `double` to store the fraction? Internally, I'm sure you'll calculate `fractionOfUnitsToCollect. numerator / fractionOfUnitsToCollect.denominator` to actually utilize it, so you're not losing any granularity and you might as well push that calculation to the client.

Reply:
Leemon has set a strict ban on all floating point calculations because different JVMs can do floating point arithmetic differently.